src/vaadin-date-time-picker.html

```diff
@@ -159,10 +236,125 @@
 
         static get observers() {
           return [
-            '__selectedDateTimeChanged(__selectedDateTime)'
+            '__selectedDateTimeChanged(__selectedDateTime)',
+            '__datePlaceholderChanged(datePlaceholder)',
+            '__timePlaceholderChanged(timePlaceholder)',
+            '__stepChanged(step)',
+            '__initialPositionChanged(initialPosition)',
+            '__showWeekNumbersChanged(showWeekNumbers)',
+            '__disabledChanged(disabled)',
+            '__readonlyChanged(readonly)'
           ];
         }
 
+        ready() {
+          super.ready();
+          this.__filterElements = node => node.nodeType === Node.ELEMENT_NODE;
+
+          this.__datePickerChanged();
+          this.__timePickerChanged();
+
+          this.$.dateSlot.addEventListener('slotchange', this.__datePickerChanged.bind(this));
+          this.$.timeSlot.addEventListener('slotchange', this.__timePickerChanged.bind(this));
+        }
+
+        __datePickerChanged() {
+          const defaultDatePicker = this.shadowRoot.querySelector('[part="date"]');
+          const assignedElements = this.$.dateSlot.assignedNodes({flatten: true}).filter(this.__filterElements);
+          const datePicker = assignedElements[0];
+          if (this.__datePicker === datePicker) {
+            return;
+          }
+          this.__datePicker = datePicker;
+
+          if (datePicker === defaultDatePicker) {
+            // Synchronize properties to default date picker
```

Comment:
Small suggestion for simplifying. We could extract something like:
```
__syncProps(to, from) {
  ['initialPosition', 'showWeekNumbers', 'disabled', 'readonly'].forEach(prop => to[prop] = from[prop]);
}
```
**Not a blocker**

Reply:
I was thinking about this, but since not all of the properties are named exactly the same on both sides (`datePlaceholder` <-> `datePicker.placeholder` and `timePlaceholder` <-> `timePicker.placeholder`) I think it's easier to understand the code having them all set in the same simple way. Otherwise it would be more complicated due to having only some of the properties set with a helper method or having more code to setup some mapping between the properties.

I prefer the current approach at least for now. Maybe if there will be many more properties later, we can think about refactoring this part.